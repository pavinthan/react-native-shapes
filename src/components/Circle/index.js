import Circle from './Circle';
import defaultProps from './defaultProps';
import propTypes from './propTypes';

Circle.propTypes = propTypes;
Circle.defaultProps = defaultProps;

export default Circle;
