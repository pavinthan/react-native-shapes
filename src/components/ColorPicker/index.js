import ColorPicker from './ColorPicker';
import defaultProps from './defaultProps';
import propTypes from './propTypes';

ColorPicker.propTypes = propTypes;
ColorPicker.defaultProps = defaultProps;

export default ColorPicker;
