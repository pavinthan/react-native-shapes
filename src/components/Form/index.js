import Form from './Form';
import defaultProps from './defaultProps';
import propTypes from './propTypes';

Form.propTypes = propTypes;
Form.defaultProps = defaultProps;

export default Form;
