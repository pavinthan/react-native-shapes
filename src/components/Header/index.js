import Header from './Header';
import defaultProps from './defaultProps';
import propTypes from './propTypes';

Header.propTypes = propTypes;
Header.defaultProps = defaultProps;

export default Header;
