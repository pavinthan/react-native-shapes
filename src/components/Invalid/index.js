import Invalid from './Invalid';
import defaultProps from './defaultProps';
import propTypes from './propTypes';

Invalid.propTypes = propTypes;
Invalid.defaultProps = defaultProps;

export default Invalid;
