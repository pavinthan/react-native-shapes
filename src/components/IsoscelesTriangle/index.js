import IsoscelesTriangle from './IsoscelesTriangle';
import defaultProps from './defaultProps';
import propTypes from './propTypes';

IsoscelesTriangle.propTypes = propTypes;
IsoscelesTriangle.defaultProps = defaultProps;

export default IsoscelesTriangle;
