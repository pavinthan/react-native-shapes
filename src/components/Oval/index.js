import Oval from './Oval';
import defaultProps from './defaultProps';
import propTypes from './propTypes';

Oval.propTypes = propTypes;
Oval.defaultProps = defaultProps;

export default Oval;
