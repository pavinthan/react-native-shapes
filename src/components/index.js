export { default as Circle } from './Circle';
export { default as ColorPicker } from './ColorPicker';
export { default as Form } from './Form';
export { default as Header } from './Header';
export { default as Invalid } from './Invalid';
export { default as IsoscelesTriangle } from './IsoscelesTriangle';
export { default as Oval } from './Oval';
export { default as Parallelogram } from './Parallelogram';
export { default as Polygon } from './Polygon';
export { default as Rectangle } from './Rectangle';
export { default as Whiteboard } from './Whiteboard';
