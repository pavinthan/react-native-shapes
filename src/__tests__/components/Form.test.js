import React from 'react';
import { Form } from '../../components';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const rendered = renderer.create(<Form />).toJSON();
  expect(rendered).toBeTruthy();
});

it('renders form component', () => {
  const tree = renderer.create(<Form />).toJSON();
  expect(tree).toMatchSnapshot();
});
