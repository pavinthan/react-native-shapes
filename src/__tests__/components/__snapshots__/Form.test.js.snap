// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders form component 1`] = `
Array [
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#000",
          "fontFamily": "System",
          "fontSize": 16,
        },
        Object {
          "fontSize": 14,
          "lineHeight": 20,
          "margin": 5,
        },
      ]
    }
    uppercase={false}
    virtual={undefined}
  >
    Draw a(n) 
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
          },
          Object {
            "color": "#d9534f",
            "fontSize": 14,
          },
        ]
      }
      uppercase={false}
      virtual={undefined}
    >
      shape
    </Text>
     with a(n)
     
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
          },
          Object {
            "color": "#d9534f",
            "fontSize": 14,
          },
        ]
      }
      uppercase={false}
      virtual={undefined}
    >
      measurement
    </Text>
     of
     
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
          },
          Object {
            "color": "#d9534f",
            "fontSize": 14,
          },
        ]
      }
      uppercase={false}
      virtual={undefined}
    >
      amount
    </Text>
    

    ( and a(n) 
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
          },
          Object {
            "color": "#d9534f",
            "fontSize": 14,
          },
        ]
      }
      uppercase={false}
      virtual={undefined}
    >
      measurement
    </Text>
     of
     
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
          },
          Object {
            "color": "#d9534f",
            "fontSize": 14,
          },
        ]
      }
      uppercase={false}
      virtual={undefined}
    >
      amount
    </Text>
     )
  </Text>,
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "borderBottomWidth": 0.5,
          "borderColor": "#ccc",
          "borderLeftWidth": 0.5,
          "borderRadius": 2,
          "borderRightWidth": 0.5,
          "borderTopWidth": 0.5,
          "elevation": 3,
          "flex": 1,
          "flexWrap": "nowrap",
          "marginBottom": 5,
          "marginLeft": 2,
          "marginRight": 2,
          "marginTop": 5,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 1.5,
        },
        Object {
          "marginBottom": 8,
        },
        undefined,
      ]
    }
    virtual={undefined}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderRadius": 2,
            "flexDirection": "row",
            "paddingBottom": 10,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 10,
          },
          Object {
            "paddingBottom": 8,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 8,
          },
        ]
      }
      virtual={undefined}
    >
      <TextInput
        allowFontScaling={true}
        error={undefined}
        multiline={true}
        onChangeText={undefined}
        placeholder="Please specify what to draw."
        placeholderTextColor="#575757"
        rowSpan={2}
        style={
          Object {
            "color": "#000",
            "fontSize": 15,
            "height": 50,
            "paddingLeft": 10,
            "paddingRight": 5,
            "textAlignVertical": "top",
          }
        }
        underlineColorAndroid="rgba(0,0,0,0)"
        virtual={undefined}
      />
    </View>
  </View>,
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": "#000",
        "borderBottomWidth": null,
        "borderColor": "#007aff",
        "borderLeftWidth": null,
        "borderRadius": 0,
        "borderRightWidth": null,
        "borderTopWidth": null,
        "elevation": 2,
        "flexDirection": "row",
        "height": 45,
        "justifyContent": "center",
        "opacity": 1,
        "paddingBottom": 6,
        "paddingTop": 6,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "#fff",
          "fontFamily": "System",
          "fontSize": 16.5,
          "marginLeft": 0,
          "marginRight": 0,
          "paddingLeft": 16,
          "paddingRight": 16,
        }
      }
      uppercase={false}
      virtual={undefined}
    >
      Draw
    </Text>
  </View>,
]
`;
